src/network/_Private/throw_socket_error.php

```diff
@@ -13,15 +13,12 @@
 use namespace HH\Lib\Experimental\{IO, Network, OS};
 use namespace HH\Lib\Str;
 
-function throw_socket_error(string $_operation, int $errno): noreturn {
-  invariant($errno !== 0, "%s should not be called on success", __FUNCTION__);
-  if ($errno < 0) {
-    throw new Network\HostResolutionException((-($errno + 10000)) as OS\HErrno);
-  }
-  switch ($errno as OS\Errno) {
-    case OS\Errno::EADDRNOTAVAIL:
-      throw new Network\AddressNotAvailableException();
-    default:
-      throw new IO\_Private\UnhandledOSErrnoException($errno as OS\Errno);
+function throw_socket_error(int $php_socket_error, string $message): noreturn {
+  invariant($php_socket_error !== 0, "%s should not be called on success", __FUNCTION__);
```

Comment:
Same here, if this somehow happens, we could at least throw some generic OS/IO exception with `$message`...?

Reply:
Both of these really feel like an appropriate usage of invariant. The error *is* that "throw_socket_error() was called by <caller> when an error didn't occur"

Changing this would mean adding 'Error: success' which always feels clowny :p

Reply:
Well this is always called with a `$message` that indicates the error, so "Error: success" can't happen but yeah if you're pretty confident that a missing errno can't happen then an invariant's fine.